lib/server/monalisa-synchronization/MonAlisaSynchronizer.js

```diff
@@ -130,41 +142,75 @@ class MonAlisaSynchronizer {
     /**
      * Fetch id and last_seen information of data pass versions that have not been deleted
      *  telling whether some data pass version must be updated
-     * @return {Object<string, { lastSeen: number, id: number }>} mapping:
+     * @return {Object<string, { lastSeen: number, id: number, status: string }>} mapping:
      * data pass version description to it's properties subset
      * @private
      */
-    async _getAllDataPassVersionsLastSeenAndId() {
+    async _getAllDataPassVersionsLastSeenAndIdAndLastStatus() {
         const dataPassesLastSeen = await DataPassVersionRepository.findAll({
             attributes: ['id', 'description', 'lastSeen'],
-            where: { deletedFromMonAlisa: false },
+            include: 'statusHistory',
+            order: [['statusHistory', 'createdAt', 'ASC']],
         });
 
-        return Object.fromEntries(dataPassesLastSeen.map(({ description, lastSeen, id }) => [description, { lastSeen, id }]));
+        return Object.fromEntries(dataPassesLastSeen.map(({ description, lastSeen, id, statusHistory }) => [
+            description,
+            { lastSeen, id, status: statusHistory.slice(-1)[0].status },
```

Comment:
I think `statusHistory[statusHistory.length - 1]` is the conventional way to go

Reply:
applied